src/main/drivers/compass/compass_lis3mdl.c

```diff
@@ -0,0 +1,164 @@
+
+// CTRL_REG1
+#define LIS3MDL_TEMP_EN             0x80  // Default 0
+#define LIS3MDL_OM_LOW_POWER        0x00  // Default
+#define LIS3MDL_OM_MED_PROF         0x20
+#define LIS3MDL_OM_HI_PROF          0x40
+#define LIS3MDL_OM_ULTRA_HI_PROF    0x60
+#define LIS3MDL_DO_0_625            0x00
+#define LIS3MDL_DO_1_25             0x04
+#define LIS3MDL_DO_2_5              0x08
+#define LIS3MDL_DO_5                0x0C
+#define LIS3MDL_DO_10               0x10  // Default
+#define LIS3MDL_DO_20               0x14
+#define LIS3MDL_DO_40               0x18
+#define LIS3MDL_DO_80               0x1C
+#define LIS3MDL_FAST_ODR            0x02
+
+// CTRL_REG2
+#define LIS3MDL_FS_4GAUSS           0x00  // Default
+#define LIS3MDL_FS_8GAUSS           0x20
+#define LIS3MDL_FS_12GAUSS          0x40
+#define LIS3MDL_FS_16GAUSS          0x60
+#define LIS3MDL_REBOOT              0x08
+#define LIS3MDL_SOFT_RST            0x04
+
+// CTRL_REG3
+#define LIS3MDL_LP                  0x20  // Default 0
+#define LIS3MDL_SIM                 0x04  // Default 0
+#define LIS3MDL_MD_CONTINUOUS       0x00  // Default
+#define LIS3MDL_MD_SINGLE           0x01
+#define LIS3MDL_MD_POWERDOWN        0x03
+
+// CTRL_REG4
+#define LIS3MDL_ZOM_LP              0x00  // Default
+#define LIS3MDL_ZOM_MP              0x04
+#define LIS3MDL_ZOM_HP              0x08
+#define LIS3MDL_ZOM_UHP             0x0C
+#define LIS3MDL_BLE                 0x02  // Default 0
+
+// CTRL_REG5
+#define LIS3MDL_FAST_READ           0x80  // Default 0
+#define LIS3MDL_BDU                 0x40  // Default 0
+
+static bool lis3mdlRead(magDev_t * mag, int16_t *magData)
+{
+    uint8_t buf[6];
+
+    busDevice_t *busdev = &mag->busdev;
+
+    bool ack = busReadRegisterBuffer(busdev, LIS3MDL_REG_OUT_X_L, buf, 6);
+
+    if (!ack) {
+        return false;
+    }
+
+    magData[X] = (int16_t)(buf[1] << 8 | buf[0]) / 4;
+    magData[Y] = (int16_t)(buf[3] << 8 | buf[2]) / 4;
+    magData[Z] = (int16_t)(buf[5] << 8 | buf[4]) / 4;
+
+    return true;
+}
+
+static bool lis3mdlInit(magDev_t *mag)
+{
+    busDevice_t *busdev = &mag->busdev;
+
+    busWriteRegister(busdev, LIS3MDL_REG_CTRL_REG2, LIS3MDL_FS_4GAUSS);
+busWriteRegister(busdev, LIS3MDL_REG_CTRL_REG1, LIS3MDL_TEMP_EN | LIS3MDL_OM_ULTRA_HI_PROF | LIS3MDL_DO_80);
+busWriteRegister(busdev, LIS3MDL_REG_CTRL_REG5, LIS3MDL_BDU);
+busWriteRegister(busdev, LIS3MDL_REG_CTRL_REG4, LIS3MDL_ZOM_UHP);
+busWriteRegister(busdev, LIS3MDL_REG_CTRL_REG3, 0x00);
+
+    delay(100);
+
+    return true;
+}
+
+bool lis3mdlDetect(magDev_t * mag)
+{
+busDevice_t *busdev = &mag->busdev;
+
+uint8_t sig = 0;
+
+    if (busdev->bustype == BUSTYPE_I2C && busdev->busdev_u.i2c.address == 0) {
+        busdev->busdev_u.i2c.address = LIS3MDL_MAG_I2C_ADDRESS;
+    }
+
+    bool ack = busReadRegisterBuffer(&mag->busdev, LIS3MDL_REG_WHO_AM_I, &sig, 1);
+
+    if (!ack || sig != LIS3MDL_DEVICE_ID) {
+//   busdev->busdev_u.i2c.address = LIS3MDL_MAG_I2C_ADDRESS2;
```

Comment:
Commented out code should:
- be removed *or*
- have a comment stating why it was commented out, and the condition of when it should be reinstated.